src/server/server_test.go

```diff
@@ -0,0 +1,86 @@
+package server
+
+import (
+	"context"
+	"fenix/src/utils"
+	"net/http"
+	"net/url"
+	"testing"
+	"time"
+
+	"github.com/gorilla/websocket"
+)
+
+func StartServer() (*utils.WaitGroupCounter, *http.Server, *ServerHub) {
+	wg := utils.NewWaitGroupCounter()
+	srv, hub := Serve("localhost:8080", wg)
+	wg.Wait()
+	return wg, srv, hub
+}
+
+func ConnectToServer(t *testing.T) *websocket.Conn {
+	u := url.URL{Scheme: "ws", Host: "localhost:8080", Path: "/ws"}
+	c, _, err := websocket.DefaultDialer.Dial(u.String(), nil)
+	if err != nil {
+		t.Fatalf("Error connecting to localhost:8080: %v", err)
+	}
+	return c
+}
+
+func TestEnsureAllGoroutinesStopWhenClientExits(t *testing.T) {
```

Comment:
Pretty lengthy test, makes no assertion. Could maybe assert that start != end.

Reply:
> Could maybe assert that start != end.

```
if start != end {
		t.Log("failed, goroutines still running:")
		t.Logf("Start: %v, End: %v", start, end)
		wg.Names.Range(
			func(key, value any) bool {
				t.Log(key)
				return true
			},
		)
		t.FailNow()
	}
	
```

> Pretty lengthy test

The test tests the actual running state of the server, so in order to set the server up correctly it is indeed lengthy.  I could take out some of the code however into a helper function.